plugins/inputs/opcua_listener/subscribe_client.go

```diff
@@ -115,9 +125,101 @@ func (sc *subscribeClientConfig) createSubscribeClient(log telegraf.Logger) (*su
 		subClient.monitoredItemsReqs[i] = req
 	}
 
+	log.Debugf("Creating event streaming items")
+	for i, node := range client.EventNodeMetricMapping {
+		req := opcua.NewMonitoredItemCreateRequestWithDefaults(node.NodeID, ua.AttributeIDValue, uint32(i))
+		if node.SamplingInterval > 0 {
+			req.RequestedParameters.SamplingInterval = float64(node.SamplingInterval)
+		}
+		filterExtObj, err := createEventFilter(client, node)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create event filter: %w", err)
+		}
+		req.RequestedParameters.Filter = filterExtObj
+		client.EventClientHandle[uint32(i)] = node
+		subClient.eventItemsReqs[i] = req
+	}
 	return subClient, nil
 }
 
+// Creation of event filter for event streaming
+func createEventFilter(client *input.OpcUAInputClient, node *input.EventNodeMetricMapping) (*ua.ExtensionObject, error) {
+
+	selects, err := createSelectClauses(node)
+	if err != nil {
+		return nil, err
+	}
+	return &ua.ExtensionObject{
+		EncodingMask: ua.ExtensionObjectBinary,
+		TypeID:       &ua.ExpandedNodeID{NodeID: ua.NewNumericNodeID(0, id.EventFilter_Encoding_DefaultBinary)},
+		Value: ua.EventFilter{
+			SelectClauses: selects,
+			WhereClause:   createWhereClauses(node),
+		},
+	}, nil
+}
```

Comment:
How about making this a member of the subscribe client to avoid passing in the client?